File: src/main/java/com/wolfbeacon/service/HackalistHackathonService.java
package com.wolfbeacon.service;

import com.wolfbeacon.model.HackalistHackathon;
import org.springframework.transaction.annotation.Transactional;

import java.util.Date;
import java.util.List;

@Transactional
public interface HackalistHackathonService {
    List<HackalistHackathon> getHackathonsBetweenDate(Date startDate, Date endDate, String sortBy, Integer count);
}

File: src/main/kotlin/wolfbeacon/hackalist/HackalistHackathonService.kt
package wolfbeacon.hackalist


import org.springframework.transaction.annotation.Transactional
import java.util.*

@Transactional
interface HackalistHackathonService {
    fun getHackathonsBetweenDatesByCoordinates(startDate: Date, endDate: Date, sortBy: String?, count: Int?, latitude: Double?, longitude: Double?): List<HackalistHackathon>
}